homeassistant/components/switch/xiaomi_miio.py

```diff
@@ -215,14 +319,30 @@ def __init__(self, name, plug, model):
         """Initialize the plug switch."""
         XiaomiPlugGenericSwitch.__init__(self, name, plug, model)
 
-        self._state_attrs = {
-            ATTR_TEMPERATURE: None,
+        if self._model == MODEL_POWER_STRIP_V2:
+            self._supported_features = SUPPORT_FLAGS_POWER_STRIP_V2
+        else:
+            self._supported_features = SUPPORT_FLAGS_POWER_STRIP_V1
+
+        self._state_attrs.update({
             ATTR_LOAD_POWER: None,
-            ATTR_MODEL: self._model,
-        }
+        })
+
+        if self.supported_features & SUPPORT_SET_POWER_MODE == 1:
```

Comment:
Should it be enough to do `if self.supported_features & SUPPORT_SET_POWER_MODE` or is this here for easier  comprehension?